post/controller.js

```diff
@@ -5,29 +5,29 @@ const express = require('express');
 const router = express.Router();
 const urlencodedParser = bodyParser.urlencoded({ extended: false });
 
-router.get(['/', '/posts'], (req, res) => {
+router.get('/posts', (req, res) => {
   postService.getAllPosts().then((posts) => {
-    res.render('posts', { posts });
+    res.json(posts);
   }).catch((err) => {
     res.status(503).send(err.message);
   });
 });
 
 router.get('/posts/:post_id', (req, res) => {
   postService.getOnePost(req.params.post_id).then((post) => {
-    res.render('post_detail', { post });
+    res.json(post);
   }).catch((err) => {
     res.status(503).send(err.message);
   });
 });
 
-router.get('/post_new', (req, res) => {
-  res.render('post_new');
-});
-
-router.post('/post_new', urlencodedParser, (req, res) => {
-  postService.createNewPost(req.body.title, req.body.author, req.body.body).then(() => {
-    res.redirect('posts');
+router.post('/posts', urlencodedParser, (req, res) => {
+  postService.createNewPost(req.body.title, req.body.author, req.body.body).then((newPost) => {
+    res.json({
+      title: newPost.title,
+      author: newPost.author,
+      body: newPost.body,
+    });
```

Comment:
definitely need to return the id here, otherwise how does the consumer know which post was just created?